infrastructure/transform/network.tf

```diff
@@ -0,0 +1,188 @@
+resource "azurerm_network_security_group" "databricks" {
+  name                = "nsg-dbks-${var.naming_suffix}"
+  resource_group_name = var.core_rg_name
+  location            = var.core_rg_location
+  tags                = var.tags
+
+  security_rule {
+    name                       = "AllowInboundDatabricksWorkerNodesToCluster"
+    description                = "Required for worker nodes communication within a cluster."
+    priority                   = 100
+    direction                  = "Inbound"
+    access                     = "Allow"
+    protocol                   = "*"
+    source_port_range          = "*"
+    destination_port_range     = "*"
+    source_address_prefix      = "VirtualNetwork"
+    destination_address_prefix = "VirtualNetwork"
+  }
+
+  security_rule {
+    name                       = "AllowOutboundDatabricksWorkerNodesToControlPlane"
+    description                = "Required for workers communication with Databricks Webapp."
+    priority                   = 100
+    direction                  = "Outbound"
+    access                     = "Allow"
+    protocol                   = "Tcp"
+    source_port_range          = "*"
+    destination_port_range     = "443"
+    source_address_prefix      = "VirtualNetwork"
+    destination_address_prefix = "AzureDatabricks"
+  }
+
+  security_rule {
+    name                       = "AllowOutboundDatabricksWorkerNodesToAzureSQLServices"
```

Comment:
Not an expert here but would this allow all connectivity from the cluster to _any_ Azure SQL database? Same for storage?

Reply:
Correct, within the virtual network that the Databricks subnets reside - one of the default rules we have to have as per https://learn.microsoft.com/en-us/azure/databricks/administration-guide/cloud-configurations/azure/vnet-inject#network-security-group-rules-for-workspaces-created-after-january-13-2020 

It does say in the docs that if certain sql instances within the network shouldn't have connectivity, to add an explicit deny on the SQL firewall